modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/api/extension/discovery/ExtensionModelLoaderRepository.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.artifact.activation.api.extension.discovery;
+
+import org.mule.runtime.extension.api.loader.ExtensionModelLoader;
+import org.mule.runtime.module.artifact.activation.internal.extension.discovery.DefaultExtensionModelLoaderRepository;
+import org.mule.runtime.module.artifact.api.plugin.LoaderDescriber;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+/**
+ * Provides access to the {@link ExtensionModelLoader} available in the container.
+ *
+ * @since 4.0, moved to api in 4.5
+ */
+@FunctionalInterface
```

Comment:
this shouldn't be a functional interface.

Reply:
Done.